Test/hofs/Twice.dfy

```diff
@@ -1,37 +0,0 @@
-// RUN: %exits-with 4 %dafny /print:"%t.print" "%s" > "%t"
-// RUN: %diff "%s.expect" "%t"
-
-function Twice<A>(f : A ~> A): A ~> A
-{
-  x requires f.requires(x) && f.requires(f(x))
-    reads f.reads(x), if f.requires(x) then f.reads(f(x)) else {}
```

Comment:
Same here, I'm surprise this example is deleted.
Is it because we can't interleave requires and reads? It seems to me we should be able to write:

```dafny
x requires f.requires(x)
   reads f.reads(x)
   requires f.requires(f(x))
   reads f.reads(f(x)) => f(f(x))
```

Reply:
Similar to `Examples.dfy`, I find it odd that `TwicePointwise` verifies, but `Twice` doesn't:

```
function Twice<A>(f : A ~> A): A ~> A {
  x requires f.requires(x) && f.requires(f(x))
    reads f.reads(x) + f.reads(f(x)) 
    => TwicePointwise(f, x)
}

function TwicePointwise<A>(f : A ~> A, x: A): A 
  requires f.requires(x) && f.requires(f(x))
  reads f.reads(x) + f.reads(f(x))
{
  f(f(x))
}
```

Why is `f.requires(x)` shared between the `requires` and `reads`-clauses of `Twicepointwise`, but not between the clauses of the lambda inside of `Twice`?

Reply:
Twice is deleted because now that the axioms for .reads and .requires have been corrected to be sound, it has revealed that there are incompleteness issues. It is not a goal of this PR to fix such issues. Instead, if we believe that it ought to verify, then we should file a distinct issue to solve this incompleteness problem.

Reply:
Fair point -- I'll approve this PR and file a separate issue.